bin/main.ml

```diff
@@ -769,10 +850,35 @@ let build_targets =
     in
     set_common common ~targets;
     let log = Log.create common in
-    Scheduler.go ~log ~common
-      (Main.setup ~log common >>= fun setup ->
-       let targets = resolve_targets_exn ~log common setup targets in
-       do_build setup targets)
+    let build_once () =
+      Main.setup ~log common
+      >>= fun setup ->
+      let targets = resolve_targets_exn ~log common setup targets in
+      do_build setup targets
+    in
+    if common.watch then
```

Comment:
This code looks very similar to the one for `dune runtest`. It can be factorized a bit more

Reply:
Done.